### PR TITLE
Initial dockerfiles

### PR DESCRIPTION
**Pull request by jgoz**
This adds two docker images that may be used to run Pli applications (back-end web service and database) and an example `docker-compose.yml` file that can be copied into a project.

It is assumed that these images will run on a recent Windows 10 or Windows Server 2019 build that includes native support for Windows containers.

Application Server:
- Runs IIS with .NET Framework (default version 4.8) and ASP.NET
- URL Rewrite module added (it took me _way too long_ to figure out that this was the reason I was seeing extremely vague 500.19 errors with the default image)
- Enabled anonymous & Windows authentication in machine config
- Added some basic debugging utilities
- Example Compose file attaches filestore as a volume bound to `${SOLUTION_ROOT}/.filestore`

Database server:
- Runs MSSQL Express on Windows Server (default 2019 LTS)
- Example Compose file attaches database from a volume bound to `${SOLUTION_ROOT}/.data`
- Connection strings:
  - When using `web` service: `Server=db,1433;Database=MyProject;User ID=sa;Password=p@ssw0rd;`
  - When connecting with management studio or local IISExpress: `Server=localhost,1433;Database=MyProject;User ID=sa;Password=p@ssw0rd;`

Basic workflow:
```sh
$ docker-compose up # run both services
$ docker-compose exec web ./debug # attach a debug monitor to the web server
$ docker-compose start db # only start the DB service (e.g., for use with local IISExpress or integration tests)
```

TODO:
- ~Readme(s)~ done
- Add to SamplePliWeb
- Add to a real project

### CODE REVIEW

File: mssql-windows/Dockerfile
Comment (101100) [inline mssql-windows/Dockerfile:6]: Why re-create their SQL image instead of building on top of it?

File: samples/pli/docker-compose.yml
Comment (101100) [inline samples/pli/docker-compose.yml:12]: While I understand this benefit for Windows devs, I don't understand how this is better for you and other non-Windows developers.  Do you run `docker-compose up` from within your windows VM, or do you use `docker-machine` to connect to a Windows host?

Did you consider putting the SQL server in Linux and the web server in a separate Windows Docker?

File: aspnet-framework/Dockerfile
Comment (101100) [inline aspnet-framework/Dockerfile:4]: Does this Dockerfile handle compilation, is the developer supposed to do the compilation first, and can this handle rebuilds?

**Review by 101100 — CHANGES_REQUESTED** [review]
Done with round 1 rom me.

File: mssql-windows/Dockerfile
Comment (jgoz) [reply mssql-windows/Dockerfile:6]: Their SQL image is built against Windows Server 2016, which cannot be run natively on Windows Server 2019.

File: samples/pli/docker-compose.yml
Comment (jgoz) [reply samples/pli/docker-compose.yml:12]: I set `DOCKER_HOST` to point to my Windows Server VM. I don't actually have Docker for Mac installed, just the CLI tools. I did consider putting SQL Server in Linux, but it's actually a worse option for me with the current limitations of Docker for Windows/Mac:
- Currently unable to run Windows Containers and LCOW simultaneously, so HyperV would be needed (which means nested virtualization, so more memory usage)
- Could split the containers and run one on the Windows VM and one on the Mac host, but this has two downsides in itself:
  - Can't use `docker-compose`
  - Docker for Mac has to use a virtualization layer to run Linux VMs, so I would essentially be running two separate virtual machines, each running a single container, which seems less efficient than a single VM with two native containers

File: aspnet-framework/Dockerfile
Comment (jgoz) [reply aspnet-framework/Dockerfile:4]: This only runs IIS with a single application at the root, bound via a docker volume mount. It doesn't handle any compilation on its own, initial or otherwise, so the workflow for the developer is the same as Start-Dev (write code, build project in IDE/command line).

It works well for me because Rider can compile the project using the Mono compiler and the output assemblies can be run natively on the .NET Framework.

File: mssql-windows/Dockerfile
Comment (101100) [reply mssql-windows/Dockerfile:6]: So Windows containers are all built in an OS version dependant way?  🤦‍♂ 

Thanks for the explanation.

File: samples/pli/docker-compose.yml
Comment (101100) [reply samples/pli/docker-compose.yml:12]: I hope that you have the time to create a skeleton README at some point.

File: aspnet-framework/Dockerfile
Comment (101100) [reply aspnet-framework/Dockerfile:4]: Oh, I didn't know you could build .NET Framework projects in Rider on non-Windows machines.  Nice.

**Review by 101100 — APPROVED** [review]
Thanks for doing the work to get started on this.  Looks good.  🍪

File: samples/pli/docker-compose.yml
Comment (jgoz) [reply samples/pli/docker-compose.yml:12]: I will do that before merging this PR - was just waiting for feedback before spending time documenting an approach that was disliked (so thank you!)

File: mssql-windows/Dockerfile
Comment (jgoz) [reply mssql-windows/Dockerfile:6]: Yes, the native windows container format requires the container image to have the same base OS as the host. This isn't true for HyperV, but I'm trying to avoid that requirement.

File: aspnet-framework/Dockerfile
Comment (jgoz) [reply aspnet-framework/Dockerfile:4]: Yeah, I just discovered that recently. I was pleasantly surprised that it worked.

**Pull request merged**
